### PR TITLE
DBZ-8208 Extend `error.processing.failure.handling.mode` to DML parsing

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-8208

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java [hunk 1]
```diff
@@ -1517,11 +1517,30 @@ protected void addToTransaction(String transactionId, LogMinerEventRow row, Supp
...
+        try {
+            event = eventSupplier.get();
+        }
+        catch (DmlParserException e) {
+            switch (connectorConfig.getEventProcessingFailureHandlingMode()) {
+                case FAIL:
+                    LOGGER.error("Failed to parse SQL for event '{}'", row);
+                    throw e;
+                case WARN:
+                    LOGGER.warn("Failed to parse SQL '{}'. The event '{}' is being ignored and skipped.", row.getRedoSql(), row);
```
Comment (thibthibus) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java:1530]:
```suggestion
                    LOGGER.warn("Failed to parse SQL for event '{}'. This event is being ignored and skipped.", row.getRedoSql(), row);
```

**Review by thibthibus — APPROVED** [review]

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java (hunk 1 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java:1530]: Hi @thibthibus, there are two arguments in this case, so you get the redo SQL when set to `WARN.`  These changes seem to ignore that fact.

Comment (thibthibus) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java:1530]: Oh my... Of course, sorry...

Comment (thibthibus) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java:1530]: Just ordered https://amzn.eu/d/daZpLfc. Apparently I need a little refresher (or to sleep more). ;-)

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java [hunk 2]
```diff
@@ -1517,11 +1517,30 @@ protected void addToTransaction(String transactionId, LogMinerEventRow row, Supp
...
         }
 
+        final LogMinerEvent event;
+        try {
+            event = eventSupplier.get();
+        }
+        catch (DmlParserException e) {
+            switch (connectorConfig.getEventProcessingFailureHandlingMode()) {
+                case FAIL:
+                    LOGGER.error("Failed to parse SQL for event '{}'", row);
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/AbstractLogMinerEventProcessor.java:1527]: Why `FAIL` and `default` branches have different message? Should not be SQL statement logged in all cases?

**Comment by jpechane** [comment]
Applied, thanks

**Comment by jpechane** [comment]
@Naros The 2.7 code is quite different. Could you please backport the change there? Is it even necessary?

**Comment by thibthibus** [comment]
> @Naros The 2.7 code is quite different. Could you please backport the change there? Is it even necessary?

It would be really helpful for us to have this change backported in the 2.7 branch. Thanks in advance.

**Pull request merged**